src/main/java/com/google/opensesame/servlets/MentorsServlet.java

```diff
@@ -60,4 +74,55 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(jsonMentors);
   }
+
+  // This function prints an error message to the mentor form page if invalid input is found.
+  public void formError(HttpServletRequest request, HttpServletResponse response, String error)
+      throws ServletException, IOException {
+    String message =
+        "<ul><li class='ml-3 display-4 font-weight-bold text-danger'>" + error + "</li></ul>";
+    response.setContentType("text/html");
+    PrintWriter pw = response.getWriter();
+    pw.print(message);
+    RequestDispatcher rd = request.getRequestDispatcher("mentor_form.html");
+    rd.include(request, response);
+    pw.close();
+    return;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+    String repoUrl = request.getParameter("inputRepo");
+    String urlPattern = "https://github.com/";
+    Pattern regEx = Pattern.compile(urlPattern);
+    Matcher inputMatcher = regEx.matcher(repoUrl);
+    if (!inputMatcher.find()) {
+      // Input is not in the format of a GitHub repo link.
+      formError(request, response, "ENTER A VALID REPO LINK");
+      return;
+    }
+
+    String repoName = repoUrl.replaceFirst("https://github.com/", "");
+    GitHub gitHub = GitHubGetter.getGitHub();
+    GHRepository inputRepo = gitHub.getRepository(repoName);
+    if (inputRepo == null) {
+      // Repo was not found on GitHub.
+      formError(request, response, "COULDN'T FIND REPO");
+      return;
+    }
+    // TODO: Send Richie the GHRepository to add to projects database
```

Comment:
When I implement my projects database function for this I think it'd be best if all of this logic is handled in the projects database function that you mention in the TODO. Keeps this function more clean and allows the code to be reused in other places if necessary. We can also talk about this on a call.